polynote-kernel/src/main/scala/polynote/kernel/Result.scala

```diff
@@ -133,19 +136,56 @@ final case class ClearResults() extends Result
 
 object ClearResults extends ResultCompanion[ClearResults](3)
 
+// TODO: fix the naming of these classes
+final case class ResultValue(
+  name: TinyString,
+  typeName: TinyString,
+  reprs: TinyList[ValueRepr],
+  sourceCell: TinyString,
+  value: Any,
+  scalaType: Universe#Type
+) extends Result
+
+object ResultValue extends ResultCompanion[ResultValue](4) {
+
+  def apply(ctx: KernelContext)(name: String, typ: Universe#Type, value: Any, sourceCell: String): ResultValue = {
+    val globalType = try typ.asInstanceOf[ctx.global.Type] catch {
+      case err: ClassCastException => throw new RuntimeException(s"Type $typ is being used in a foreign kernel", err)
+    }
```

Comment:
I think there might be a couple places where we do this conversion, maybe KernelContext should have a `pathifyType` method (definitely use a better name than that) that does this conversion?